### PR TITLE
Add format module and implement buildifier transform to format Bazel/Starlark files

### PR DESCRIPTION
**Pull request by Yannic**
This PR adds a new transformation that runs [buildifier](https://github.com/bazelbuild/buildtools/tree/master/buildifier) during import/export.

Related: #98

Note: This is still WIP (name of binary hardcoded, no tests, ...). I would appreciate an early high-level review to make sure the overall design of this is good, though.

### CODE REVIEW

**Comment by hsudhof** [comment]
I will take a look as soon as I can, thank you for the contribution!

File: java/com/google/copybara/transform/BUILD [hunk 1]
```diff
@@ -39,5 +39,6 @@ java_library(
         "//third_party:jsr305",
         "//third_party:re2j",
         "//third_party:skylark-lang",
+        "//third_party/bazel:shell",
```
Comment (hsudhof) [inline java/com/google/copybara/transform/BUILD]:
```suggestion
        "//third_party:shell",
```

**Comment by hsudhof** [comment]
One question up front: Do you plan to add tests?

**Comment by hsudhof** [comment]
We have something very similar internally, I will see  if we can combine the two implementations in OSS.

**Comment by Yannic** [comment]
> One question up front: Do you plan to add tests?

I plan to add at least integration tests to ensure the transform works as aspected.
Haven't really looked at how a unittest for this could look like, but I'm open to add some if they are reasonable easy to write.

> We have something very similar internally, I will see if we can combine the two implementations in OSS.

Sounds good.

**Comment by hsudhof** [comment]
Thank you for sending this - we appreciate the contribution! 
We have a lot of internal usage, so we have limited maneuvering space, but I feel that some aspects of your PR would make a great addition - in particular using the mode flag. The question is whether it could be the same transform or maybe a format.buildozer_check to make it more readable. 

Cheers,
~H

**Pull request closed without merging**